oak_runtime/src/node/http/tests.rs

```diff
@@ -507,3 +527,226 @@ async fn send_request(
+    // channel. A handle to the read half of the channel is returned. The Oak node will use it
+    // to fetch the write handle to an invocation channel for sending the invocations to the
+    // HTTP client pseudo-node.
+    let oak_node_init_receiver =
+        create_client_node(&runtime, "www.google.com".to_string(), UriScheme::Https);
+
+    // Create a sync_channel to be notified when the Oak Node is completed.
+    let (result_sender, result_receiver) = mpsc::sync_channel(1);
+
+    // Create a test Oak node that sends requests to the HTTP client pseudo-node.
+    let client_test_node = ClientTesterNode {
+        uri: "https://www.google.com/".to_string(),
+        result_sender,
+    };
+
+    // Register the test Oak node in the runtime.
+    runtime
+        .node_register(
+            Box::new(client_test_node),
+            "client_test_node",
+            &Label::public_untrusted(),
+            oak_node_init_receiver,
+        )
+        .unwrap();
+
+    // Wait for the test Node to complete execution before terminating the Runtime.
+    let result_value = result_receiver.recv().expect("test node disconnected");
+
+    assert!(result_value.is_ok());
+    let resp = result_value.unwrap();
+    // The test uses a self signed certificate. So, we expect an error that is converted to a 500
+    // response in the HTTP client node.
+    assert_eq!(resp.status, 500);
+    runtime.runtime.stop();
+}
+
+#[test]
+fn test_https_client_can_handle_http_requests_to_an_external_service() {
+    init_logger();
+
+    let runtime = create_runtime();
+    // This is a public node, so the authority can be empty.
+    let empty_authority = "".to_string();
+    // Create an HTTP client pseudo-node to serve `HTTP` requests`.
+    let oak_node_init_receiver = create_client_node(&runtime, empty_authority, UriScheme::Http);
+
+    // Create a sync_channel to be notified when the Oak Node is completed.
+    let (result_sender, result_receiver) = mpsc::sync_channel(1);
+
+    // Create a test Oak node that sends requests to the HTTP client pseudo-node.
+    let client_test_node = ClientTesterNode {
+        uri: "http://www.google.com".to_string(),
+        result_sender,
+    };
+
+    // Register the test Oak node in the runtime.
+    runtime
+        .node_register(
+            Box::new(client_test_node),
+            "client_test_node",
+            &Label::public_untrusted(),
+            oak_node_init_receiver,
+        )
+        .unwrap();
+
+    // Wait for the test Node to complete execution before terminating the Runtime.
+    let result_value = result_receiver.recv().expect("test node disconnected");
+
+    assert!(result_value.is_ok());
+    let resp = result_value.unwrap();
+    assert_eq!(resp.status, 200);
+    runtime.runtime.stop();
+}
+
+/// Creates an HTTP client pseudo-node in the given Runtime.
+fn create_client_node(
+    runtime: &RuntimeProxy,
+    authority: String,
+    scheme: UriScheme,
```

Comment:
It may read better if scheme came first, I was confused when reading the invocation and didn't see that HTTPS was after the domain name.